app/assets/javascripts/exchanges/carriers.js

```diff
@@ -0,0 +1,70 @@
+$(document).on('turbolinks:load', function() {
+    $('.filter-form button[type="submit"]').on('click', function(event) {
+        event.preventDefault();
+
+        filterPlans();
+    });
+
+    $('#clear-filters').on('click', function() {
+        $('.filter-form input[type="checkbox"]').prop('checked', false);
+        $('.filter-form input[type="text"]').val('');
+
+        filterPlans();
+    });
+
+    function filterPlans() {
+        const planTypes = [];
+        const pvpRatingAreas = [];
+        const metalLevels = [];
+        const searchPlan = $('input[name="search"]').val().trim().toLowerCase();
+
+        $('input[name="plan_type[]"]:checked').each(function() {
+            planTypes.push($(this).val().toLowerCase());
+        });
+
+        $('input[name="pvp_rating_areas[]"]:checked').each(function() {
+            pvpRatingAreas.push($(this).val().toLowerCase());
+        });
+
+        $('input[name="metal_level[]"]:checked').each(function() {
+            metalLevels.push($(this).val().toLowerCase());
+        });
+
+        let visiblePlansCount = 0;
+
+        $('tbody tr').each(function() {
```

Comment:
Might be safer to put some qualifying class/id in here that the `tbody` is searched for under (e.g, `#carrier-plans tbody tr` or something?). Who knows when a new table could land in the page/template and break this logic.